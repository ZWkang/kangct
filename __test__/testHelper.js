// import React from 'react';
import { mount, shallow, render } from 'enzyme';

function setUp(type, ...arg) {
  const [Component, props = {}, children = ''] = arg;
  if (~['mount', 'shallow', 'render'].indexOf(type)) {
    return global[type](<Component {...props}>{children}</Component>);
  }
}

export function AutoMath(Component, props = {}, children = '') {
  const wrapper = setUp('render', Component, props, children);
  expect(wrapper).toMatchSnapshot();
}

export function ComponentRenderWithoutError(Component, props = {}, children = '') {
  const wrapper = setUp('mount', Component, props, children);
  expect(() => {
    wrapper.setProps({});
    wrapper.unmount();
  }).not.toThrow();
}

export function childrenShouldBeProps(Component, children) {
  const wrapper = setUp('mount', Component, {}, children);

  expect(wrapper.get(0).props.children).toEqual(children);
}

export function setup(Component, props) {
  const wrapper = setUp(Component, props, null);
  return {
    wrapper,
    props
  };
}
