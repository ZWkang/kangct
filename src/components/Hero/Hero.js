import React from 'react';
import styled from 'styled-components';
import is from 'styled-is';
const Hero = styled.section`
  margin-left: auto;
  margin-right: auto;
  display: flex;
  align-items: center;
`;

const HeroLeft = styled.div`
  flex: 1;
`;
Hero.HeroLeft = HeroLeft;

const HeroH1 = styled.h1`
  margin-bottom: 0.4em;

  font-size: 35px;
  font-weight: 600;
  line-height: 58px;
  letter-spacing: -0.2px;
`;
Hero.HeroH1 = HeroH1;

const HeroP = styled.p`
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin-bottom: 40px;
`;

// & a.Link {
const HeroA = styled.a`
  color: rgb(66, 90, 112);
  border-bottom: 2px solid rgba(66, 90, 112, 0.05);

  &:hover {
    border-bottom-color: var(--green-base);
  }

  &:active {
    color: rgba(66, 90, 112, 0.1);
    border-bottom-color: rgba(66, 90, 112, 0.1);
  }
`;
Hero.HeroA = HeroA;

const HeroRight = styled.div`
  flex: 1;
  padding-left: 48px;
  & img {
    width: 100%;
  }
`;
Hero.HeroRight = HeroRight;

const HeroBody = styled.div`
  display: flex;
  align-items: center;
  padding: 4em 2em;
  ${is('medium')`
    padding: 5em 2.5em;
  `}
  ${is('large')`
    padding: 6em 3em;
  `}
  margin-right: 0;
  margin-left: 0;
  max-width: 1024px;

  /* -webkit-box-align: center;
    align-items: center;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    margin-left: auto;
    margin-right: auto;
    
    padding: 80px 24px; */
`;
Hero.HeroBody = HeroBody;
export { HeroBody, HeroA, Hero, HeroH1, HeroLeft, HeroRight, HeroP };
