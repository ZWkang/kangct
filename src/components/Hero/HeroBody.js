import React, { PureComponent } from 'react';
import styled from 'styled-components';
import is from 'styled-is';
// class HeroBody extends PureComponent {
//   constructor(props) {}
//   render() {
//       return (

//       )
//   }
// }
