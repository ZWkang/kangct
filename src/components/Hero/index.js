export {
  Hero as default,
  HeroBody,
  HeroA,
  HeroContainer,
  HeroH1,
  HeroLeft,
  HeroRight,
  HeroP
} from './Hero';
