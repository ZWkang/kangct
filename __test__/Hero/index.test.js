// import React from 'react';
// import { mount, shallow, render } from 'enzyme';
import Hero from '@app/Hero/index.js';
import { AutoMath, ComponentRenderWithoutError } from '../testHelper';
import 'jest-styled-components';
import { mount } from 'enzyme';

// https://github.com/airbnb/enzyme/issues/1798 dive vs shallow
describe('Alert', () => {
  describe('behavior test', () => {
    it('render without error', () => {
      ComponentRenderWithoutError(Hero);
    });
    it('test props snapshot', () => {
      AutoMath(
        () => {
          return (
            <Hero>
              <Hero.HeroBody>
                <Hero.HeroLeft>
                  <Hero.HeroH1>hero</Hero.HeroH1>
                </Hero.HeroLeft>
                <Hero.HeroRight>
                  <img src="https://cdn.pixabay.com/photo/2016/09/02/08/32/cuba-1638594_960_720.jpg" />
                </Hero.HeroRight>
              </Hero.HeroBody>
            </Hero>
          );
        },
        {},
        'here is a test alert'
      );
    });
  });
  describe('HeroBody behavior test', () => {
    it('render without error', () => {
      ComponentRenderWithoutError(Hero.HeroBody);
    });
    it('test props snapshot', () => {
      AutoMath(Hero.HeroBody, { medium: true, large: false }, 'here is a test alert');
    });

    it('medium size', () => {
      const Component = <Hero.HeroBody medium={true} />;
      const mountedComponent = mount(Component);

      expect(mountedComponent.find(Hero.HeroBody)).toHaveStyleRule('padding', '5em 2.5em');
    });

    it('large size', () => {
      const Component = <Hero.HeroBody large={true} />;
      const mountedComponent = mount(Component);

      expect(mountedComponent.find(Hero.HeroBody)).toHaveStyleRule('padding', '6em 3em');
    });
  });
});
