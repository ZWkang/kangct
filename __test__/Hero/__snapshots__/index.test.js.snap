// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert HeroBody behavior test test props snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-bZQynM eZdFSC",
    },
    "children": Array [
      Object {
        "data": "here is a test alert",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Alert behavior test test props snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-bdVaJa fJuDDm",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "sc-bZQynM faqDcQ",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "sc-bwzfXH kMDzjb",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "sc-htpNat faLytS",
                },
                "children": Array [
                  Object {
                    "data": "hero",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "h1",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "sc-EHOje cIDLxn",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "src": "https://cdn.pixabay.com/photo/2016/09/02/08/32/cuba-1638594_960_720.jpg",
                  },
                  "children": Array [],
                  "name": "img",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "src": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "src": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "sc-EHOje cIDLxn",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "src": "https://cdn.pixabay.com/photo/2016/09/02/08/32/cuba-1638594_960_720.jpg",
                },
                "children": Array [],
                "name": "img",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "src": undefined,
                },
                "x-attribsPrefix": Object {
                  "src": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "sc-bwzfXH kMDzjb",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "sc-htpNat faLytS",
                  },
                  "children": Array [
                    Object {
                      "data": "hero",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "h1",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "section",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
